com.unity.render-pipelines.universal/Editor/ShaderGraph/Targets/UniversalTarget.cs

```diff
@@ -686,19 +686,17 @@ static class CoreKeywordDescriptors
         public static readonly KeywordDescriptor MainLightShadows = new KeywordDescriptor()
         {
             displayName = "Main Light Shadows",
-            referenceName = "_MAIN_LIGHT_SHADOWS",
-            type = KeywordType.Boolean,
-            definition = KeywordDefinition.MultiCompile,
-            scope = KeywordScope.Global,
-        };
-
-        public static readonly KeywordDescriptor MainLightShadowsCascade = new KeywordDescriptor()
-        {
-            displayName = "Main Light Shadows Cascade",
-            referenceName = "_MAIN_LIGHT_SHADOWS_CASCADE",
-            type = KeywordType.Boolean,
+            referenceName = "",
+            type = KeywordType.Enum,
             definition = KeywordDefinition.MultiCompile,
             scope = KeywordScope.Global,
+            entries = new KeywordEntry[]
+            {
+                new KeywordEntry() { displayName = "Off", referenceName = "" },
+                new KeywordEntry() { displayName = "No Cascade", referenceName = "MAIN_LIGHT_SHADOWS" },
```

Comment:
Does shader graph prepend the missing `_` here?

Reply:
Yes, shader graph add automatically the '_' in front of all keywords. If the missing '_' was added to the referenceName, the name would be like '__MAIN_LIGHT_SHADOWS'.